libgnucash/backend/dbi/gnc-backend-dbi.cpp

```diff
@@ -283,34 +293,41 @@ GncDbiBackend<Type>::conn_setup (PairVec& options, UriStrings& uri)
 template <DbType Type>bool
 GncDbiBackend<Type>::create_database(dbi_conn conn, const char* db)
 {
-    const char *dbname;
+    const char *dbname = nullptr;
     const char *dbcreate;
     if (Type == DbType::DBI_MYSQL)
     {
-        dbname = "mysql";
         dbcreate = "CREATE DATABASE %s CHARACTER SET utf8";
+	clear_options(conn, {"dbname"});
     }
     else
     {
         dbname = "postgres";
         dbcreate = "CREATE DATABASE %s WITH TEMPLATE template0 ENCODING 'UTF8'";
-    }
-    PairVec options;
-    options.push_back(std::make_pair("dbname", dbname));
-    try
-    {
-        set_options(conn, options);
-    }
-    catch (std::runtime_error& err)
-    {
-        set_error (ERR_BACKEND_SERVER_ERR);
-        return false;
+	PairVec options;
+	options.push_back(std::make_pair("dbname", dbname));
+	try
+	{
+	    set_options(conn, options);
+	}
+	catch (std::runtime_error& err)
+	{
+	    set_error (ERR_BACKEND_SERVER_ERR);
+	    return false;
+	}
```

Comment:
This might be overcome by switching to `information_schema` for both backends, but it occurs to me that it MySQL and Postgresql have different behaviors it would be cleaner to have separate specializations.

Reply:
I suppose it would be "constructive" if I would first provide a PR with information_schema. If that works then it would be a very tiny change -- just connect to the info-schema instead of the privileged root dbs.

Question would also be: for MySQL: would it be sufficient to support MySQL-server versions released after 2007, roughly 18 years ago?

For Postgres: I seldom use it, should check for the support of information_schema .. todo.

There are two scenarios for a to-be-written PR with info-schema:

1. minimal: just replace the name of the root-db with "information_schema"
2. with fallback: try "information_schema", if that fails fall back to the current state of the affair.

Reply:
> MySQL-server versions released after 2007, roughly 18 years ago?

Sure. GnuCash didn't support MySQL until 2011 (2.4.0 introduced it and was released 20 December 2010). But I'd be comfortable with whatever was oldest in the stable versions of the main Linux distros (Arch, Debian/Ubuntu, Fedora, Gentoo, OpenSuSE, and Slackware) as of 2020. We just have to document what it is.

> For Postgres: I seldom use it, should check for the support of information_schema .. todo.

Note that the CI runners aren't set up to do this but there are a [couple of environment variables](https://github.com/Gnucash/gnucash/blob/stable/libgnucash/backend/dbi/test/test-backend-dbi.cpp#L34) that if set with valid credentials will run the DBI backend tests against MySQL and Postgresql.